images/upload-gitlab-failure-logs/upload_gitlab_failure_logs.py

```diff
@@ -1,13 +1,46 @@
-from datetime import datetime
 import json
 import os
 import re
+from datetime import datetime
 from pathlib import Path
+from time import sleep
+from typing import Any
 
 import gitlab
 import yaml
+from kubernetes import client, config
+from kubernetes.client.exceptions import ApiException
+from kubernetes.client.models.v1_pod import V1Pod
+from kubernetes.client.models.v1_pod_status import V1PodStatus
 from opensearch_dsl import Date, Document, connections
 
+config.load_config()
+v1_client = client.CoreV1Api()
+
+
+def get_pod_status(pod_name: str) -> dict[str, Any]:
+    pod: V1Pod | None = None
+    while True:
+        # Try to fetch pod with kube
+        try:
+            pod = v1_client.read_namespaced_pod(name=pod_name, namespace="pipeline")
+        except ApiException:
+            # If it doesn't work, that means the pod has already been cleaned up.
+            # In that case, we break out of the loop and return.
+            break
+
+        # Check if the pod is still running. If so, keep re-fetching it until it's complete
```

Comment:
In the cases we've seen where pods stick around forever, does this have a chance of blocking up the error processing system completely until we "unstuck" the pod?

Reply:
Or will another instance of the failure processor be triggered by the web event, and only the one waiting for the stuck pods status will be hung?  Or maybe I misunderstand.

Reply:
The latter - the "stuck job" error processing job will continue to run while waiting for the stuck pod to be killed by the stuck pod controller, while any new job failures will continue to have new jobs kicked off for them.